en/lessons/basics/control-structures.md

```diff
@@ -1,207 +1,246 @@
-...>   {:error} -> "Uh oh!"
-...>   _ -> "Catch all"
-...> end
-"Hello World"
+```
+result = if (boolExpression) 
+         then (expression1)
+         else (expression2)
 ```
 
-The `_` variable is an important inclusion in `case/2` statements. Without it, failure to find a match will raise an error:
+- If the first expression evaluates to `True`, then the expression after the
+  `then` keyword is evaluated
+- if it evaluates to `False`, then the expression after the `else` keyword is
+  evaluated
+
+### Examples
+
+```haskell
+-- Is this integer even or odd?
+evenOrOdd :: Int -> Text
+evenOrOdd val = 
+    if val `mod` 2 == 0 then "Value is even" else "Value is odd"
+
+-- How to navigate a traffic light
+trafficLight :: Text -> Text
+trafficLight lightColor = 
+    if (lightColor == "Red" || lightColor == "Yellow")
+    then "Slow down"
+    else "Roll through"
+
+-- Give a nice description for the weather
+-- Note how `warmOrCold` and `niceOrGloomy` are defined in a let block then 
+-- used in the final definition
+weatherDescription :: Text -> Int -> Text
+weatherDescription weather tempInC = let
+    warmOrCold   = if tempInC > 21       then "It's warm" else "It's cold"
+    niceOrGloomy = if weather == "sunny" then "nice out"  else "gloomy out"
+    in
+    warmOrCold <> " and " <> niceOrGloomy
+```
 
-```elixir
-iex> case :even do
-...>   :odd -> "Odd"
-...> end
-** (CaseClauseError) no case clause matching: :even
+### Usage
 
-iex> case :even do
-...>   :odd -> "Odd"
-...>   _ -> "Not Odd"
-...> end
-"Not Odd"
-```
+`if` statements are not the most common control structure 
 
-Consider `_` as the `else` that will match "everything else".
+They are handy if you have a simple definition, and don't need one of the more
+flexible structures we will see soon
 
-Since `case/2` relies on pattern matching, all of the same rules and restrictions apply.
-If you intend to match against existing variables you must use the pin `^/1` operator:
+## Case Statements
 
-```elixir
-iex> pie = 3.14
- 3.14
-iex> case "cherry pie" do
-...>   ^pie -> "Not so tasty"
-...>   pie -> "I bet #{pie} is tasty"
-...> end
-"I bet cherry pie is tasty"
-```
+Case statements offer a way to switch on any one of a number of possible values
 
-Another neat feature of `case/2` is its support for guard clauses:
+### Syntax
 
-_This example comes directly from the official Elixir [Getting Started](https://elixir-lang.org/getting-started/case-cond-and-if.html#case) guide._
+```
+result = case (expression) of
+    (pattern1) -> (expression1) 
+    (pattern2) -> (expression2)
+    ...
+    _          -> (expressionLast)
+```
 
-```elixir
-iex> case {1, 2, 3} do
-...>   {1, x, 3} when x > 0 ->
-...>     "Will match"
-...>   _ ->
-...>     "Won't match"
-...> end
-"Will match"
+- The expression between the `case` and `of` keywords is evaluated
+- Its result is looked up in the list of patterns
+- For the first pattern that matches, its expression is evaluated
```

Comment:
Wouldn't be nice to me more explicit: "The first pattern that matches -- and only it -- has its expression evaluated"?